src/main/java/exercises/nguonchhay/OopExercise.java

```diff
@@ -0,0 +1,315 @@
+	protected String updatedBy;
+	protected long state;
+	protected long version;
+	
+	public DCommon() {}
+	
+	public DCommon(long id, Date createdDate, Date updatedDate, String createdBy, String updatedBy, long state, long version) {
+		setId(id);
+		setCreatedDate(createdDate);
+		setUpdatedDate(updatedDate);
+		setCreatedBy(createdBy);
+		setUpdatedBy(updatedBy);
+		setState(state);
+		setVersion(version);
+	}
+	
+	public long getId() {
+		return id;
+	}
+	
+	public void setId(long id) {
+		this.id = id;
+	}
+	
+	public Date getCreatedDate() {
+		return createdDate;
+	}
+	
+	public String getCreateDateString() {
+		if (null == createdDate) {
+			return "";
+		}
+		return createdDate.toString();
+	}
+	
+	public void setCreatedDate(Date createdDate) {
+		this.createdDate = createdDate;
+	}
+	
+	public Date getUpdatedDate() {
+		return updatedDate;
+	}
+	
+	public void setUpdatedDate(Date updatedDate) {
+		this.updatedDate = updatedDate;
+	}
+	
+	public String getUpdatedDateString() {
+		if (null == updatedDate) {
+			return "";
+		}
+		return updatedDate.toString();
+	}
+	
+	public String getCreatedBy() {
+		return createdBy;
+	}
+	
+	public void setCreatedBy(String createdBy) {
+		this.createdBy = createdBy;
+	}
+	
+	public String getUpdatedBy() {
+		return updatedBy;
+	}
+	
+	public void setUpdatedBy(String updatedBy) {
+		this.updatedBy = updatedBy;
+	}
+	
+	public long getState() {
+		return state;
+	}
+	
+	public void setState(long state) {
+		this.state = state;
+	}
+	
+	public long getVersion() {
+		return version;
+	}
+	
+	public void setVersion(long version) {
+		this.version = version;
+	}
+}
+
+/**
+ * Person Class
+ */
+class Person extends DCommon {
+	protected Date birthDate;
+	protected String lastName;
+	protected String firstName;
+	
+	public Person() {}
+	
+	public Person(String lastName, String firstName, Date birthDate) {
+		setLastName(lastName);
+		setFirstName(firstName);
+		setBirthDate(birthDate);
+	}
+	
+	public Person(long id, String lastName, String firstName, Date birthDate, Date createdDate, Date updatedDate, String createdBy, String updatedBy, long state, long version) {
+		super(id, createdDate, updatedDate, createdBy, updatedBy, state, version);
+		setLastName(lastName);
+		setFirstName(firstName);
+		setBirthDate(birthDate);
+	}
+	
+	public Date getBirthDate() {
+		return birthDate;
+	}
+	
+	public void setBirthDate(Date birthDate) {
+		this.birthDate = birthDate;
+	}
+	
+	public String getBirthDateString() {
+		if (null == birthDate) {
+			return "";
+		}
+		return birthDate.toString();
+	}
+	
+	public String getLastName() {
+		return lastName;
+	}
+	
+	public void setLastName(String lastName) {
+		this.lastName = lastName;
+	}
+	
+	public String getFirstName() {
+		return firstName;
+	}
+	
+	public void setFirstName(String firstName) {
+		this.firstName = firstName;
+	}
+	
+	public String getFullname() {
+		return new StringBuffer(lastName + " " + firstName).toString().toUpperCase();
```

Comment:
may be it should not convert toUpperCase(), let use this outside rather than in getFullName(); 
if you want to do this, method name should be getFullNameInUpperCase()